certbot-postfix/certbot_postfix/tests/installer_test.py

```diff
@@ -0,0 +1,316 @@
+        installer.Installer.add_parser_arguments(mock_add)
+
+        for call in mock_add.call_args_list:
+            self.assertTrue(call[0][0] in options)
+
+    def test_no_postconf_prepare(self):
+        with create_installer(self.config) as installer:
+            installer_path = "certbot_postfix.installer"
+            exe_exists_path = installer_path + ".certbot_util.exe_exists"
+            path_surgery_path = "certbot_postfix.util.plugins_util.path_surgery"
+            with mock.patch(path_surgery_path, return_value=False):
+                with mock.patch(exe_exists_path, return_value=False):
+                    self.assertRaises(errors.NoInstallationError,
+                                      installer.prepare)
+
+    def test_old_version(self):
+        with create_installer(self.config, main_cf=_main_cf_with({"mail_version": "0.0.1"}))\
+                as installer:
+            self.assertRaises(errors.NotSupportedError, installer.prepare)
+
+    def test_lock_error(self):
+        with create_installer(self.config) as installer:
+            assert_raises = functools.partial(self.assertRaises,
+                                              errors.PluginError,
+                                              installer.prepare)
+            certbot_test_util.lock_and_call(assert_raises, self.tempdir)
+
+
+    @mock.patch('certbot.util.lock_dir_until_exit')
+    def test_dir_locked(self, lock_dir):
+        with create_installer(self.config) as installer:
+            lock_dir.side_effect = errors.LockError
+            self.assertRaises(errors.PluginError, installer.prepare)
+
+    def test_more_info(self):
+        with create_installer(self.config) as installer:
+            installer.prepare()
+            output = installer.more_info()
+            self.assertTrue("Postfix" in output)
+            self.assertTrue(self.tempdir in output)
+            self.assertTrue(DEFAULT_MAIN_CF["mail_version"] in output)
+
+    def test_get_all_names(self):
+        config = {"mydomain": "example.org",
+                  "myhostname": "mail.example.org",
+                  "myorigin": "example.org"}
+        with create_installer(self.config, main_cf=_main_cf_with(config)) as installer:
+            installer.prepare()
+            result = installer.get_all_names()
+            self.assertEqual(result, set(config.values()))
+
+    @certbot_test_util.patch_get_utility()
+    def test_deploy(self, mock_util):
+        mock_util().yesno.return_value = True
+        from certbot_postfix import constants
+        with create_installer(self.config) as installer:
+            installer.prepare()
+
+            # pylint: disable=protected-access
+            installer.deploy_cert("example.com", "cert_path", "key_path",
+                                  "chain_path", "fullchain_path")
+            changes = installer.postconf.get_changes()
+            expected = {} # type: Dict[str, Union[Tuple[str, str], str]]
```

Comment:
I think the `Tuple` here should be `Tuple[str, ...]` because the tuple can have an arbitrary number of values.

Also, are you planning on always using tuples in `constants.py` for single values? I think that's fine, but if you are I think you can remove the `Union` here.